### PR TITLE
Fix: running more than once with -rename=false will corrupt hcl file

### LINKED ISSUE #88: Running more than once with -rename=false will corrupt hcl file
### Steps to reproduce
1. Run Terratag with `-rename=false`
2. Run it again with `-rename=false`

### Expected
Terratag should update the previously generated locals block

### Actual
Terratag adds an additional locals block with the same key, causing terraform to fail validation:  

```hcl
locals {
  localTagKey  = "localTagKey"
  localTagKey2 = "localTagKey2"
  localMap = {
    key = "value"
  }
}
locals {
  terratag_added_main = { "some_tag" = "value" }
}


locals {
  terratag_added_main = { "some_tag" = "value" }
}
```

### PR DESCRIPTION
**Pull request by TomerHeber**
fixes #88.
Avoids corruption and duplication when running a formatted file.

### CODE REVIEW

File: test/fixture/terraform_15_1.0/git_issue_88__corrupt_hcl_file/expected/main.terratag.tf
Comment (TomerHeber) [inline test/fixture/terraform_15_1.0/git_issue_88__corrupt_hcl_file/expected/main.terratag.tf:12]: This is still WIP. The test will fail because of this line.
It will modify this line and apply a merge to the merge.

File: test/fixture/terraform_15_1.0/git_issue_88__corrupt_hcl_file/expected/main.terratag.tf
Comment (TomerHeber) [inline test/fixture/terraform_15_1.0/git_issue_88__corrupt_hcl_file/expected/main.terratag.tf:30]: This test expects no double locals, and instead a merged locals. This was fixed.

File: convert/convert.go
Comment (TomerHeber) [inline convert/convert.go:122]: See the comment. But basically before appending a new merged terratag locals, remove the old one (if exist).

File: main.go
Comment (TomerHeber) [inline main.go:133]: This idea is to get the existing locals, merge it with terratag.Added. 
Remove the old locals, and append the new (merged) one.

**Comment by TomerHeber** [comment]
Note: I expect integration tests to fail!
This is still a work in progress. However, I wanted some feedback before I fix the existing issue. (see comments in added integration test).

File: convert/locals.go
Comment (razbensimon) [inline convert/locals.go:12]: Super complicated regex.
Just wondering what example of text it matches so I can understand better why we need that.

**Review by razbensimon — COMMENTED** [review]
@TomerHeber generally looks good to me.
I'm not sure why we need to encode and decode but maybe I miss something.

I'm not so familiar with this code but maybe others can give better input here 
@eranelbaz @yaronya @RLRabinowitz

File: convert/locals.go
Comment (TomerHeber) [reply convert/locals.go:12]: @razbensimon 
It looks for "key" = "value" entries

Generally, locals may look like this (over complicating it on purpose):
```
{"a" = "b",      "c"=  "d" ,
  "g" = "f" }
```

Comment (razbensimon) [reply convert/locals.go:12]: can it contain a more complex type than string?

Comment (TomerHeber) [reply convert/locals.go:12]: we're specifically parsing the terratag locals. Therefore, it shouldn't. But who knows what folks would do with their files...

File: convert/convert.go
Comment (RLRabinowitz) [reply convert/convert.go:122]: Something I might be missing here.
This code basically creates a terratag `locals` block, and remove the old terratag `locals` block beforehand if it already exists.

So:

1. This isn't really an `AppendLocalsBlock`, right? More of a "reset" locals block
2. There's more code here that merges non-terratag `locals` block with terratag `locals` block. Is that necessary?

File: main.go
Comment (RLRabinowitz) [inline main.go:128]: We already have code that overwrites the terratag `locals` block
Seems like this code merges existing `locals` blocks with the terratag one.

Is that necessary?

File: convert/locals.go
Comment (RLRabinowitz) [reply convert/locals.go:12]: The regex seems wrong to me. Both key and value of `locals` don't have to include quotes. See https://www.terraform.io/language/values/locals

Also a minor point, we should probably allow for tabs as well as spaces instead of using `[ ]*`

File: main.go
Comment (RLRabinowitz) [inline main.go:137]: If I'm not mistaken, this won't work if user has more than 1 `locals` block (not terratag-generated)

**Review by RLRabinowitz — COMMENTED** [review]
Left some comments, seems like most of the code revolves around making sure there is only one `locals` block, which isn't a must for this feature (might be a nice-to-have though)

File: convert/convert.go
Comment (TomerHeber) [reply convert/convert.go:122]: @razbensimon - you're correct about 1. I think some names changes are due.
About 2: No, the code doesn't merge non-terratag locals (it's not necessary and probably a bad idea). It only merges the terratag locals. This is to fix issues such as a corrupted HCL if running terratag more than once. (again probably naming is confusing and will be fixed).

File: convert/locals.go
Comment (TomerHeber) [reply convert/locals.go:12]: You're assuming that it's supposed to handle other locals (which is not the case - as explained above) this should only handle the terratag locals, which are in this style.
Happy to add \t as well.

File: main.go
Comment (TomerHeber) [reply main.go:128]: I don't believe there's code that merges existing Terratag locals. Where did you see it? (This is the code that I'm writing at the moment).

File: main.go
Comment (TomerHeber) [reply main.go:137]: Sorry for repeating myself. This only works on the terratag local. It will skip all other locals.

File: convert/convert.go
Comment (RLRabinowitz) [reply convert/convert.go:122]: We've talked, I've misunderstood, thought the purpose here was to specifically overwrite the terratag local, and not merge it with previous values. That's a good solution as well. Let's go with that 👍🏻

File: convert/locals.go
Comment (RLRabinowitz) [reply convert/locals.go:12]: Regarding the locals part, I misunderstood, though this had to do with all locals

File: main.go
Comment (RLRabinowitz) [reply main.go:128]: I misunderstood, summary [here](https://github.com/env0/terratag/pull/118/files#r854117741)

File: main.go
Comment (RLRabinowitz) [reply main.go:137]: I misunderstood. Summary [here](https://github.com/env0/terratag/pull/118/files#r854117741)

File: convert/terratag_locals.go
Comment (TomerHeber) [inline convert/terratag_locals.go:17]: Made some naming changes to help clarify this only applies to the terratag locals.

File: test/fixture/terraform_15_1.0/git_issue_88__corrupt_hcl_file/expected/main.terratag.tf
Comment (TomerHeber) [inline test/fixture/terraform_15_1.0/git_issue_88__corrupt_hcl_file/expected/main.terratag.tf:12]: Notice this line. It's a marge of a merge. This is still a valid HCL file. Regardless, I will open a new ticket in case it's something to be fixed along the road.

File: convert/terratag_locals.go
Comment (TomerHeber) [inline convert/terratag_locals.go:13]: added tabs to the regex as well. (just in case).

**Review by RLRabinowitz — APPROVED** [review]
Looks good 👍🏻

**Pull request merged**
